### ISSUE TITLE
Updating MyPaint's PKGBUILDs for AUR

### ISSUE THREAD

**Issue by odysseywestra** [mypaint/mypaint#831]
I'm working on updating the PKGBUILD file for [mypaint-git](https://aur.archlinux.org/packages/mypaint-git/) for the Arch User Repository. My end goal is to get both the stable versions and git versions to be able to be installed side-by-side and remove the dependency on gegl untill mypaint/libmypaint#97 is resolved. So I basically I just need someone to double check my work since this is my first time making a PKGBUILD file. The file is based on @achadwick's [PKGBUILD](https://github.com/Alexpux/MINGW-packages/blob/master/mingw-w64-mypaint-git/PKGBUILD) for mingw package, mypaint-git.

```bash
# Maintainer: Sapphira Armageddos <shadowkyogre.public@gmail.com>
# Maintainer: Albert Westra <odysseywestra@gmail.com>
# Contributor: Jon Nordby <jononor@gmail.com>
# Contributor: mosra <mosra@centrum.cz>

pkgname=mypaint-git
pkgver=1.3.0.alpha+git.922f73df
pkgrel=1
pkgdesc="Simple drawing & painting program that works well with graphics tablets (git)."
arch=('i686' 'x86_64')
url="http://mypaint.org/"
license=('GPL2')
depends=('desktop-file-utils' 'gtk3' 'json-c' 'lcms2' 'librsvg' 'python2-cairo' 'python2-gobject' 'python2-numpy' 'gsettings-desktop-schemas' 'libmypaint-git')
makedepends=('git' 'swig' 'pygobject-devel' 'python2' 'python2-setuptools')
provides=('${pkgname}')
conflicts=('${pkgname}')
install=${pkgname}.install
source=('git+https://github.com/mypaint/mypaint.git'
        '${pkgname}.install')
sha256sums=('SKIP')

pkgver() {
    cd "${srcdir}/${pkgname}"
    eval `python2 lib/meta.py`
    echo "$MYPAINT_VERSION_CEREMONIAL" \
    | sed 's/^v//;s/\([^-]*-g\)/r\1/;s/-/./g' \
    | sed 's/gitexport/git/g'
}

prepare() {
    cd "${srcdir}/${pkgname}"
}

build() {
    cd "${srcdir}/${pkgname}"
    python2 setup.py clean --all
    python2 setup.py build
}

package() {
    cd "${srcdir}/${pkgname}"
    echo "root: ${pkgdir}"
	
    python2 setup.py install \
    --root="${pkgdir}" --skip-build
	
    install -Dm644 COPYING \
        "${pkgdir}/share/licenses/${pkgname}/COPYING"
    install -Dm644 Licenses.md \
        "${pkgdir}/share/licenses/${pkgname}/Licenses.md"
    install -Dm644 Licenses.dep5 \
        "${pkgdir}/share/licenses/${pkgname}/Licenses.dep5"
}
```

Info on packaging for Arch: https://wiki.archlinux.org/index.php/Creating_packages
Arch User Repository: https://wiki.archlinux.org/index.php/Arch_User_Repository

I'm also going to be updating also PKGBUILDs for [libmypaint-git](https://aur.archlinux.org/packages/libmypaint-git/) by removing the gegl dependency on the AUR.

**Comment by ncmprhnsbl** [comment]
i'm **not** a PKGBUILD expert but have found some issues...
after making some changes, have managed to build 1.3.0-alpha+git.922f73df (that works!)
using libmypaint-git built without the gegl option (build fails otherwise)

so, heres the issues:
``==> ERROR: ${pkgname}.install was not found in the build directory and is not a URL.``
not sure why this is, but simply changing ``${pkgname}`` to ``mypaint-git`` fixes it

``==> ERROR: Integrity checks (sha256) differ in size from the source array.``
copied the sha256 from AUR PKGBUILD seems to fix it

```==> Starting prepare()...
/home/monkey/Build/mypaint-git/PKGBUILD: line 32: cd: /home/monkey/Build/mypaint-git/src/mypaint-git: No such file or directory
```
needs to be src/mypaint  so changing ``${pkgname}`` to ``mypaint`` in the relevant lines fixes it

also provides and conflicts should be ``mypaint`` otherwise mypaint-git conflicts with itself  ;p

here's what i used:   (note: theres undoubtably  more elegant solutions)
```
# Maintainer: Sapphira Armageddos <shadowkyogre.public@gmail.com>
# Maintainer: Albert Westra <odysseywestra@gmail.com>
# Contributor: Jon Nordby <jononor@gmail.com>
# Contributor: mosra <mosra@centrum.cz>

pkgname=mypaint-git
pkgver=1.3.0.alpha+git.922f73df
pkgrel=1
pkgdesc="Simple drawing & painting program that works well with graphics tablets (git)."
arch=('i686' 'x86_64')
url="http://mypaint.org/"
license=('GPL2')
depends=('desktop-file-utils' 'gtk3' 'json-c' 'lcms2' 'librsvg' 'python2-cairo' 'python2-gobject' 'python2-numpy' 'gsettings-desktop-schemas' 'libmypaint-git')
makedepends=('git' 'swig' 'pygobject-devel' 'python2' 'python2-setuptools')
provides=('mypaint')
conflicts=('mypaint')
install=mypaint-git.install  
source=('git+https://github.com/mypaint/mypaint.git'
       'mypaint-git.install'
)

pkgver() {
    cd "${srcdir}/mypaint"
    eval `python2 lib/meta.py`
    echo "$MYPAINT_VERSION_CEREMONIAL" \
    | sed 's/^v//;s/\([^-]*-g\)/r\1/;s/-/./g' \
    | sed 's/gitexport/git/g'
}

prepare() {
    cd "${srcdir}/mypaint"
}

build() {
    cd "${srcdir}/mypaint"
    python2 setup.py clean --all
    python2 setup.py build
}

package() {
    cd "${srcdir}/mypaint"
    echo "root: ${pkgdir}"
	
    python2 setup.py install \
    --root="${pkgdir}" --skip-build
	
    install -Dm644 COPYING \
        "${pkgdir}/share/licenses/${pkgname}/COPYING"
    install -Dm644 Licenses.md \
        "${pkgdir}/share/licenses/${pkgname}/Licenses.md"
    install -Dm644 Licenses.dep5 \
        "${pkgdir}/share/licenses/${pkgname}/Licenses.dep5"
}
sha256sums=('SKIP'
            '34dec9df2e005de9fae6c6352f49835a1d41e515c6131ed1df8c9e27ee3c9d1d')
```

**Comment by odysseywestra** [comment]
Awesome sauce, I came up to the same solution. I was trying to come up with a way to install both stable and master side by side, but for now I'm just going to make sure it installs with setuptools. I'll worry about that later.

With libmypaint, yeah all I had to do was remove the gegl dependency and it build and installed. Here what I have for libmypaint-git

```
# Maintainer: Iru Cai <mytbk920423@gmail.com>

_basever=2.0.0.alpha
pkgname=libmypaint-git
pkgver=2.0.0.alpha+git.1672d09
pkgrel=1
pkgdesc="a library for making brushstrokes which is used by MyPaint and other projects, with GEGL support"
arch=('i686' 'x86_64')
url="http://mypaint.org/"
license=('ISC')
depends=('gobject-introspection' 'json-c')
makedepends=('git' 'intltool' 'python')
provides=("libmypaint=${pkgver}")
conflicts=('libmypaint')
source=('git+https://github.com/mypaint/libmypaint.git')
md5sums=('SKIP')

build() {
  cd "$srcdir/libmypaint"

  ./autogen.sh
  ./configure --prefix=/usr
  make
}

package() {
  cd "$srcdir/libmypaint"

  make DESTDIR="$pkgdir" install
  install -D -m644 COPYING "$pkgdir/usr/share/licenses/libmypaint/LICENSE"
}

pkgver() {
  cd "$srcdir/libmypaint"

  printf "${_basever}+git.%s" "$(git rev-parse --short HEAD)" | sed 's/^v//g;s/-/./g'
}

# vim:set ts=2 sw=2 et:
```

**Comment by ncmprhnsbl** [comment]
had another look at it and made some refinements:
```
# Maintainer: Sapphira Armageddos <shadowkyogre.public@gmail.com>
# Maintainer: Albert Westra <odysseywestra@gmail.com>
# Contributor: Jon Nordby <jononor@gmail.com>
# Contributor: mosra <mosra@centrum.cz>



pkgrel=1
pkgdesc="Simple drawing & painting program that works well with graphics tablets (git)."
arch=('i686' 'x86_64')
url="http://mypaint.org/"
license=('GPL2')
depends=('desktop-file-utils' 'gtk3' 'json-c' 'lcms2' 'librsvg' 'python2-cairo' 'python2-gobject' 'python2-numpy' 'gsettings-desktop-schemas' 'libmypaint-git')
makedepends=('git' 'swig' 'pygobject-devel' 'python2' 'python2-setuptools')
provides=('mypaint')
conflicts=('mypaint')
install=mypaint-git.install  
source=('git+https://github.com/mypaint/mypaint.git'
       'mypaint-git.install'
)


pkgver() {
    cd "${srcdir}/${_pkgname}"
    eval `python2 lib/meta.py`
    echo "$MYPAINT_VERSION_CEREMONIAL" \
    | sed 's/^v//;s/\([^-]*-g\)/r\1/;s/-/./g' \
    | sed 's/gitexport/git/g'
}

prepare() {
    cd "${srcdir}/${_pkgname}"
}

build() {
    cd "${srcdir}/${_pkgname}"
    python2 setup.py clean --all
    python2 setup.py build
}

package() {
    cd "${srcdir}/${_pkgname}"
    echo "root: ${pkgdir}"
	
    python2 setup.py install \
    --root="${pkgdir}" --skip-build
	
    install -Dm644 COPYING \
        "${pkgdir}/share/licenses/${pkgname}/COPYING"
    install -Dm644 Licenses.md \
        "${pkgdir}/share/licenses/${pkgname}/Licenses.md"
    install -Dm644 Licenses.dep5 \
        "${pkgdir}/share/licenses/${pkgname}/Licenses.dep5"
}
sha256sums=('SKIP'
            '34dec9df2e005de9fae6c6352f49835a1d41e515c6131ed1df8c9e27ee3c9d1d')
```
using the ``_pkgname=mypaint`` variable
and  
```
SRC=https://github.com/mypaint/mypaint
COM=`git ls-remote $SRC refs/heads/master | cut -c 1-8`
pkgver=1.3.0.alpha+git.$COM
```
automatically fetchs the latest commit number for the pkgver..
(havnt tried a full build this way, but preliminary checks seem fine)
don't know whether this plays nice with the AUR versioning tho...

am i correct in assming that ``libmypaint`` built without gegl won't work for gimp-2.9 builds?

**Comment by odysseywestra** [comment]
That is being addressed in mypaint/libmypaint#97 and I informed the package maintainer for gimp-git that they can switch back to using libmypaint-git if they so wish.

Okay I've updated both mypaint-git and the libmypaint-git, so I'm going to close this issue. If anyone has an issue with the AUR packages, just either leave a comment on the package page, or open up an issue here and tag me.

**Comment by achadwick** [comment]
@ncmprhnsbl 
> am i correct in assming that libmypaint built without gegl won't work for gimp-2.9 builds?

I don't believe so rn, but I am seeking confirmation on the #gimp IRC. I was told an age back that GIMP did not need gegl-enabled MyPaint.

We need to know whether GEGL support is required by GIMP, because if so it needs to be part of the standard libmypaint build for things like AUR and MSYS2, not turned off by default.
